### PR TITLE
bug: Docker compose supporting multiple files

### PR DESCRIPTION
**Pull request by mariabelenvivanco**
As a response to  #1782 the docker_compose() function on the tiltfile should support multiple files.  
i.e: 
`docker_compose(["docker-compose1.yml", "docker-compose2.yml"]`

### CODE REVIEW

File: internal/dockercompose/client.go [hunk 1]
```diff
@@ -83,10 +92,14 @@ func (c *cmdDCClient) Down(ctx context.Context, configPath string, stdout, stder
...
-func (c *cmdDCClient) StreamLogs(ctx context.Context, configPath string, serviceName model.TargetName) (io.ReadCloser, error) {
+func (c *cmdDCClient) StreamLogs(ctx context.Context, configPath []string, serviceName model.TargetName) (io.ReadCloser, error) {
 	// TODO(maia): --since time
 	// (may need to implement with `docker log <cID>` instead since `d-c log` doesn't support `--since`
-	args := []string{"-f", configPath, "logs", "-f", "-t", serviceName.String()}
+	var args []string
+	for _, config := range configPath {
+		args = append(args, "-f", config)
+	}
+	args = append(args, "logs", "-f", "-t", serviceName.String())
```
Comment (jazzdan) [inline internal/dockercompose/client.go]: Hmm what is this `"-f"` doing here?

File: internal/dockercompose/client.go [hunk 2]
```diff
@@ -140,21 +157,22 @@ func (c *cmdDCClient) StreamEvents(ctx context.Context, configPath string) (<-ch
 		err = cmd.Wait()
 		if err != nil {
 			logger.Get(ctx).Infof("[DOCKER-COMPOSE WATCHER] exited with error: %v", err)
+			logger.Get(ctx).Infof("command running %v", cmd)
```
Comment (jazzdan) [inline internal/dockercompose/client.go]: 👀

File: internal/dockercompose/client.go [hunk 3]
```diff
@@ -140,21 +157,22 @@ func (c *cmdDCClient) StreamEvents(ctx context.Context, configPath string) (<-ch
...
 }
 
-func (c *cmdDCClient) Services(ctx context.Context, configPath string) (string, error) {
-	return c.dcOutput(ctx, configPath, "config", "--services")
+func (c *cmdDCClient) Services(ctx context.Context, configPaths []string) (string, error) {
+	return c.dcOutput(ctx, configPaths, "config", "--services")
 }
 
-func (c *cmdDCClient) ContainerID(ctx context.Context, configPath string, serviceName model.TargetName) (container.ID, error) {
+func (c *cmdDCClient) ContainerID(ctx context.Context, configPath []string, serviceName model.TargetName) (container.ID, error) {
```
Comment (jazzdan) [inline internal/dockercompose/client.go]:
```suggestion
func (c *cmdDCClient) ContainerID(ctx context.Context, configPaths []string, serviceName model.TargetName) (container.ID, error) {
```

File: internal/dockercompose/client.go [hunk 4]
```diff
@@ -169,8 +187,13 @@ func (c *cmdDCClient) dcCommand(ctx context.Context, args []string) *exec.Cmd {
 	return cmd
 }
 
-func (c *cmdDCClient) dcOutput(ctx context.Context, configPath string, args ...string) (string, error) {
-	args = append([]string{"-f", configPath}, args...)
+func (c *cmdDCClient) dcOutput(ctx context.Context, configPath []string, args ...string) (string, error) {
```
Comment (jazzdan) [inline internal/dockercompose/client.go]:
```suggestion
func (c *cmdDCClient) dcOutput(ctx context.Context, configPaths []string, args ...string) (string, error) {
```

File: internal/dockercompose/fake_client.go [hunk 5]
```diff
@@ -40,17 +40,17 @@ func NewFakeDockerComposeClient(t *testing.T, ctx context.Context) *FakeDCClient
 	}
 }
 
-func (c *FakeDCClient) Up(ctx context.Context, pathToConfig string, serviceName model.TargetName,
+func (c *FakeDCClient) Up(ctx context.Context, pathToConfig []string, serviceName model.TargetName,
```
Comment (jazzdan) [inline internal/dockercompose/fake_client.go]:
```suggestion
func (c *FakeDCClient) Up(ctx context.Context, configPaths []string, serviceName model.TargetName,
```

File: internal/dockercompose/fake_client.go [hunk 6]
```diff
@@ -40,17 +40,17 @@ func NewFakeDockerComposeClient(t *testing.T, ctx context.Context) *FakeDCClient
...
 	shouldBuild bool, stdout, stderr io.Writer) error {
 	c.UpCalls = append(c.UpCalls, UpCall{pathToConfig, serviceName, shouldBuild})
 	return nil
 }
 
-func (c *FakeDCClient) Down(ctx context.Context, pathToConfig string, stdout, stderr io.Writer) error {
+func (c *FakeDCClient) Down(ctx context.Context, pathToConfig []string, stdout, stderr io.Writer) error {
```
Comment (jazzdan) [inline internal/dockercompose/fake_client.go]:
```suggestion
func (c *FakeDCClient) Down(ctx context.Context, configPaths []string, stdout, stderr io.Writer) error {
```

File: internal/dockercompose/fake_client.go [hunk 7]
```diff
@@ -40,17 +40,17 @@ func NewFakeDockerComposeClient(t *testing.T, ctx context.Context) *FakeDCClient
...
 	return nil
 }
 
-func (c *FakeDCClient) StreamLogs(ctx context.Context, pathToConfig string, serviceName model.TargetName) (io.ReadCloser, error) {
+func (c *FakeDCClient) StreamLogs(ctx context.Context, pathToConfig []string, serviceName model.TargetName) (io.ReadCloser, error) {
```
Comment (jazzdan) [inline internal/dockercompose/fake_client.go]:
```suggestion
func (c *FakeDCClient) StreamLogs(ctx context.Context, configPaths []string, serviceName model.TargetName) (io.ReadCloser, error) {
```

File: internal/dockercompose/fake_client.go [hunk 8]
```diff
@@ -72,7 +72,7 @@ func (c *FakeDCClient) StreamLogs(ctx context.Context, pathToConfig string, serv
 	return reader, nil
 }
 
-func (c *FakeDCClient) StreamEvents(ctx context.Context, pathToConfig string) (<-chan string, error) {
+func (c *FakeDCClient) StreamEvents(ctx context.Context, pathToConfig []string) (<-chan string, error) {
```
Comment (jazzdan) [inline internal/dockercompose/fake_client.go]:
```suggestion
func (c *FakeDCClient) StreamEvents(ctx context.Context, configPaths []string) (<-chan string, error) {
```

File: internal/dockercompose/fake_client.go [hunk 9]
```diff
@@ -102,14 +102,14 @@ func (c *FakeDCClient) SendEvent(evt Event) error {
...
 	return c.ConfigOutput, nil
 }
 
-func (c *FakeDCClient) Services(ctx context.Context, pathToConfig string) (string, error) {
+func (c *FakeDCClient) Services(ctx context.Context, configPaths []string) (string, error) {
 	return c.ServicesOutput, nil
 }
 
-func (c *FakeDCClient) ContainerID(ctx context.Context, pathToConfig string, serviceName model.TargetName) (container.ID, error) {
+func (c *FakeDCClient) ContainerID(ctx context.Context, pathToConfig []string, serviceName model.TargetName) (container.ID, error) {
```
Comment (jazzdan) [inline internal/dockercompose/fake_client.go]:
```suggestion
func (c *FakeDCClient) ContainerID(ctx context.Context, configPaths []string, serviceName model.TargetName) (container.ID, error) {
```

File: internal/engine/upper_test.go [hunk 10]
```diff
@@ -2044,6 +2044,7 @@ func TestDockerComposeUp(t *testing.T) {
 	assert.Equal(t, server.DockerComposeTarget().ID(), call.dc().ID())
 }
 
+//TODO (maria) tests for dockercompose mulltiple files
```
Comment (jazzdan) [inline internal/engine/upper_test.go]: 👀

File: internal/hud/view/view.go [hunk 11]
```diff
@@ -17,16 +17,16 @@ type ResourceInfoView interface {
...
-	ConfigPath      string
+	ConfigPaths     []string
 	ContainerStatus dockercompose.Status
 	ContainerID     container.ID
 	Log             model.Log
 	StartTime       time.Time
 }
 
-func NewDCResourceInfo(configPath string, status dockercompose.Status, cID container.ID, log model.Log, startTime time.Time) DCResourceInfo {
+func NewDCResourceInfo(configPath []string, status dockercompose.Status, cID container.ID, log model.Log, startTime time.Time) DCResourceInfo {
```
Comment (jazzdan) [inline internal/hud/view/view.go]:
```suggestion
func NewDCResourceInfo(configPaths []string, status dockercompose.Status, cID container.ID, log model.Log, startTime time.Time) DCResourceInfo {
```

File: internal/tiltfile/docker_compose.go [hunk 12]
```diff
@@ -23,34 +23,50 @@ import (
...
 
 func (s *tiltfileState) dockerCompose(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
-	var configPath string
-	err := starlark.UnpackArgs(fn.Name(), args, kwargs, "configPath", &configPath)
+	var configPathsValue starlark.Value
+
+	err := starlark.UnpackArgs(fn.Name(), args, kwargs, "configPaths", &configPathsValue)
 	if err != nil {
 		return nil, err
+	} // parsing args
```
Comment (jazzdan) [inline internal/tiltfile/docker_compose.go]: Is this comment necessary?

File: internal/tiltfile/docker_compose.go [hunk 13]
```diff
@@ -330,7 +346,7 @@ func (c dcConfig) GetService(name string) (dcService, error) {
 	return svc, nil
 }
 
-func serviceNames(ctx context.Context, dcc dockercompose.DockerComposeClient, configPath string) ([]string, error) {
+func serviceNames(ctx context.Context, dcc dockercompose.DockerComposeClient, configPath []string) ([]string, error) {
```
Comment (jazzdan) [inline internal/tiltfile/docker_compose.go]:
```suggestion
func serviceNames(ctx context.Context, dcc dockercompose.DockerComposeClient, configPaths []string) ([]string, error) {
```

File: internal/tiltfile/docker_compose.go [hunk 14]
```diff
@@ -350,7 +366,8 @@ func serviceNames(ctx context.Context, dcc dockercompose.DockerComposeClient, co
 	return result, nil
 }
 
-func parseDCConfig(ctx context.Context, dcc dockercompose.DockerComposeClient, configPath string) ([]*dcService, error) {
+func parseDCConfig(ctx context.Context, dcc dockercompose.DockerComposeClient, configPath []string) ([]*dcService, error) {
```
Comment (jazzdan) [inline internal/tiltfile/docker_compose.go]:
```suggestion
func parseDCConfig(ctx context.Context, dcc dockercompose.DockerComposeClient, configPaths []string) ([]*dcService, error) {
```

File: internal/tiltfile/docker_compose.go [hunk 15]
```diff
@@ -369,13 +386,15 @@ func parseDCConfig(ctx context.Context, dcc dockercompose.DockerComposeClient, c
 }
 
 func getConfigAndServiceNames(ctx context.Context, dcc dockercompose.DockerComposeClient,
-	configPath string) (conf dcConfig, svcNames []string, err error) {
+	configPath []string) (conf dcConfig, svcNames []string, err error) {
```
Comment (jazzdan) [inline internal/tiltfile/docker_compose.go]:
```suggestion
	configPaths []string) (conf dcConfig, svcNames []string, err error) {
```

File: internal/tiltfile/docker_compose.go [hunk 16]
```diff
@@ -400,12 +419,12 @@ func getConfigAndServiceNames(ctx context.Context, dcc dockercompose.DockerCompo
 	return conf, svcNames, err
 }
 
-func (s *tiltfileState) dcServiceToManifest(service *dcService, dcConfigPath string) (manifest model.Manifest,
+func (s *tiltfileState) dcServiceToManifest(service *dcService, dcConfigPath []string) (manifest model.Manifest,
```
Comment (jazzdan) [inline internal/tiltfile/docker_compose.go]:
```suggestion
func (s *tiltfileState) dcServiceToManifest(service *dcService, configPaths []string) (manifest model.Manifest,
```

File: internal/tiltfile/docker_compose.go [hunk 17]
```diff
@@ -426,7 +445,11 @@ func (s *tiltfileState) dcServiceToManifest(service *dcService, dcConfigPath str
 
 	dcInfo = dcInfo.WithBuildPath(service.BuildContext)
 
-	paths := []string{path.Dir(service.DfPath), path.Dir(dcConfigPath)}
+	var paths []string
```
Comment (jazzdan) [inline internal/tiltfile/docker_compose.go]:
```suggestion
	paths := []string{path.Dir(service.Dfpath)}
```

Then you can delete L452.

File: internal/tiltfile/tiltfile_docker_compose_test.go [hunk 18]
```diff
@@ -61,7 +61,7 @@ ports:
 - 12312:80/tcp`, f.JoinPath("foo"))
 }
 
-func TestDockerComposeManifest(t *testing.T) {
+func TestDockerComposeManifest(t *testing.T) { //TODO MARIA
```
Comment (jazzdan) [inline internal/tiltfile/tiltfile_docker_compose_test.go]: 👀

File: internal/tiltfile/tiltfile_docker_compose_test.go [hunk 19]
```diff
@@ -518,10 +518,10 @@ func (f *fixture) assertDcManifest(name string, opts ...interface{}) model.Manif
 }
 
 type dcConfigPathHelper struct {
-	path string
+	path []string
```
Comment (jazzdan) [inline internal/tiltfile/tiltfile_docker_compose_test.go]:
```suggestion
	paths []string
```

File: internal/tiltfile/tiltfile_docker_compose_test.go [hunk 20]
```diff
@@ -518,10 +518,10 @@ func (f *fixture) assertDcManifest(name string, opts ...interface{}) model.Manif
...
 }
 
-func dcConfigPath(path string) dcConfigPathHelper {
+func dcConfigPath(path []string) dcConfigPathHelper {
```
Comment (jazzdan) [inline internal/tiltfile/tiltfile_docker_compose_test.go]:
```suggestion
func dcConfigPath(paths []string) dcConfigPathHelper {
```

File: internal/tiltfile/tiltfile_state.go [hunk 21]
```diff
@@ -976,9 +977,11 @@ func (s *tiltfileState) translateDC(dc dcResourceSet) ([]model.Manifest, error)
 		// e.g. dc.yml specifies one Dockerfile but the imageTarget specifies another
 		s.configFiles = sliceutils.DedupedAndSorted(append(s.configFiles, configFiles...))
 	}
-	if dc.configPath != "" {
-		s.configFiles = sliceutils.DedupedAndSorted(append(s.configFiles, dc.configPath))
+	if dc.configPaths != nil {
```
Comment (jazzdan) [inline internal/tiltfile/tiltfile_state.go]:
```suggestion
	if len(dc.configPaths) != 0 {
```

File: internal/dockercompose/client.go (hunk 1 above)
Comment (mariabelenvivanco) [reply internal/dockercompose/client.go]: The first -f in the for loop specifies a file to compose as seen in the docker compose documentation: 
image


The second -f in the append method serves as an option of the logs command to "follow"  as seen in the docker compose logs documentation: 
image

File: internal/tiltfile/tiltfile_docker_compose_test.go (hunk 18 above)
Comment (mariabelenvivanco) [reply internal/tiltfile/tiltfile_docker_compose_test.go]: I'm sorry i keep doing this

Comment (jazzdan) [reply internal/tiltfile/tiltfile_docker_compose_test.go]: No worries! 😄

**Comment by maiamcc** [comment]
Oh no, I think this introduced a bug! 😮 

On master:
```
$ tilt down
Beginning Tiltfile execution
Successfully loaded Tiltfile
Deleting via kubectl: Deployment/sail


        Can't find a suitable configuration file in this directory or any
        parent. Are you in the right directory?

        Supported filenames: docker-compose.yml, docker-compose.yaml

error running `docker-compose down`: command ["docker-compose" "down"] failed.
error: exit status 1
```

With your change reverted:
```
$ tilt down
Beginning Tiltfile execution
Successfully loaded Tiltfile
Deleting via kubectl: Deployment/sail
```

**Pull request merged**